src/main/java/cubicchunks/worldgen/dependency/RegionDependency.java

```diff
@@ -36,40 +36,24 @@
 public class RegionDependency implements Dependency {
 	
 	private GeneratorStage targetStage;
-	
-	private Set<Long> requiredCubes;
-	
 	private Set<Requirement> requirements;
-
-	private int xLow;
-	private int xHigh;
-	private int yLow;
-	private int yHigh;
-	private int zLow;
-	private int zHigh;
 	
-	
-	public RegionDependency(GeneratorStage targetStage, int radius) {
+	public RegionDependency(Cube cube, GeneratorStage targetStage, int radius) {
```

Comment:
The proposed flexibility is already there. The GeneratorPipeline calls `GeneratorStage.getDependency(cube) (== DependencyProvider.getDependency(cube)`
It does not matter if the stage uses a singleton or an individual instance per cube, because the calls to Dependency.getRequirements and DependencyManager.update receive the cube as parameter. I've changed RegionDependency to act as a Singleton as you suggested.